### PR TITLE
dts spi cleanup

### PR DESCRIPTION
**Pull request by galak**
Pull in some commits from @tbursztyka and update them a little for various other cleanups related to SPI controllers now all supporting DTS.

### CODE REVIEW

**Comment by superna9999** [comment]
Ok for the STM32 changes

File: boards/arm/frdm_k64f/dts_fixup.h [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
-#define CONFIG_IEEE802154_MCR20A_SPI_DRV_NAME		DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_BUS_NAME
-#define CONFIG_IEEE802154_MCR20A_SPI_SLAVE		DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_BASE_ADDRESS
-#define CONFIG_IEEE802154_MCR20A_SPI_FREQ		DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_SPI_MAX_FREQUENCY
-#define CONFIG_IEEE802154_MCR20A_GPIO_SPI_CS_DRV_NAME	DT_NXP_KINETIS_DSPI_4002C000_CS_GPIOS_CONTROLLER
-#define CONFIG_IEEE802154_MCR20A_GPIO_SPI_CS_PIN	DT_NXP_KINETIS_DSPI_4002C000_CS_GPIOS_PIN
-#define CONFIG_MCR20A_GPIO_IRQ_B_NAME			DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_IRQB_GPIOS_CONTROLLER
-#define CONFIG_MCR20A_GPIO_IRQ_B_PIN			DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_IRQB_GPIOS_PIN
-#define CONFIG_MCR20A_GPIO_RESET_NAME			DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_CONTROLLER
-#define CONFIG_MCR20A_GPIO_RESET_PIN			DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_PIN
+#define DT_IEEE802154_MCR20A_SPI_DRV_NAME		DT_NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_BUS_NAME
```
Comment (nashif) [inline boards/arm/frdm_k64f/dts_fixup.h]: should we move those out of the board? frdm_k64f does not have an MCR20A AFAIK.

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/arm/frdm_k64f/dts_fixup.h (hunk 1 above)
Comment (galak) [reply boards/arm/frdm_k64f/dts_fixup.h]: Yes, can we do it as a follow up PR.  Because soc/arm/nxp_kinetis/kwx/dts_fixup.h also needs addressing.

Comment (nashif) [reply boards/arm/frdm_k64f/dts_fixup.h]: > soc/arm/nxp_kinetis/kwx/dts_fixup.h

afaik this guy actually has a radio

Comment (nashif) [reply boards/arm/frdm_k64f/dts_fixup.h]: @MaureenHelm is that correct?

Comment (galak) [reply boards/arm/frdm_k64f/dts_fixup.h]: Now that you mention it, that sounds familiar.

Comment (galak) [reply boards/arm/frdm_k64f/dts_fixup.h]: Either way, the pull out of MCR20A from the frdm_k64f.dts I'll handle in a followup PR.

Comment (galak) [reply boards/arm/frdm_k64f/dts_fixup.h]: We have issue #7462 for it.

Comment (nashif) [reply boards/arm/frdm_k64f/dts_fixup.h]: ok

**Review by nashif — APPROVED** [review]

**Pull request merged**